### PR TITLE
Improve OkHttp documentation.

### PR DESCRIPTION
**Pull request by swankjesse**
Combine the marketing page (index.md) with the GitHub README.md.

Fix links pointing to 3.x documentation.

Use Markdown tables instead of HTML tables.

### CODE REVIEW

File: BUG-BOUNTY.md [hunk 1]
```diff
@@ -1,10 +0,0 @@
-Serious about security
-======================
-
-Square recognizes the important contributions the security research community
-can make. We therefore encourage reporting security issues with the code
-contained in this repository.
-
-If you believe you have discovered a security vulnerability, please follow the
-guidelines at https://hackerone.com/square-open-source
```
Comment (yschimke) [inline BUG-BOUNTY.md]: This contradicts the hackerone.com page which specifically tests for this file

https://hackerone.com/square-open-source

Comment (swankjesse) [reply BUG-BOUNTY.md]: Oooh, good catch. Lemme revert.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,72 @@
...
+```
+
+
+Post to a Server
+----------------
+
+This program posts data to a service. [Full source][post_example].
+
+```java
+public static final MediaType JSON
```
Comment (yschimke) [inline README.md]: We should fix this in 4.1, provide well known MediaTypes

Comment (swankjesse) [reply README.md]: I am not a fan. Guava’s list seems like a massive hassle.

Comment (JakeWharton) [reply README.md]: And they begrudgingly have to keep adding more and more endlessly...

Comment (swankjesse) [reply README.md]: https://github.com/google/guava/search?q=MediaType&type=Commits

Comment (yschimke) [reply README.md]: I just want JSON :)

Comment (swankjesse) [reply README.md]: Yeah, but I don’t trust you to stop there, and I can’t provide a pithy rule for what we include and what we exclude.

Comment (yschimke) [reply README.md]: Rule of 3, so we should add application/vnd.openxmlformats-officedocument.spreadsheetml.sheet now

**Comment by swankjesse** [comment]
Merging docs changes optimistically! “LGTM” Reviews still welcome.

**Pull request merged**